File: python/examples/tool_router/direct_tools_preset.py
"""
Tool Router direct_tools session preset with OpenAI Agents.

session_preset="direct_tools" is a shortcut for sessions where the full allowed
tool set is known upfront. It disables Tool Router meta/helper tools by default
and loads all tools allowed by the filters directly into session.tools() and the
MCP tool list.

Usage:
    COMPOSIO_API_KEY=... OPENAI_API_KEY=... python examples/tool_router/direct_tools_preset.py
"""

import os

from agents import Agent, Runner
from composio_openai_agents import OpenAIAgentsProvider

from composio import Composio


def require_env(name: str) -> str:
    value = os.environ.get(name)
    if not value:
        raise RuntimeError(f"Set {name} before running this example.")
    return value


composio = Composio(
    api_key=require_env("COMPOSIO_API_KEY"),
    provider=OpenAIAgentsProvider(),
)
require_env("OPENAI_API_KEY")

session = composio.create(
    user_id="direct-tools-example-user",
    session_preset="direct_tools",
    toolkits=["hackernews"],
    tools={"hackernews": {"enable": ["HACKERNEWS_GET_USER"]}},
)

tools = session.tools()
print("Direct tools exposed to the agent:")
for tool in tools:
    print(f"- {tool.name}")

agent = Agent(
    name="Direct Tools Demo Agent",
    instructions=(
        "Use the provided direct tools. The session is configured without Tool "
        "Router meta/helper tools."
    ),
    model=os.environ.get("OPENAI_MODEL", "gpt-4.1-mini"),
    tools=tools,
)

result = Runner.run_sync(
    agent,
    'Use the Hacker News tool to look up user "pg" and report their karma.',
)
print(result.final_output)

File: ts/examples/tool-router/src/direct-tools-preset.ts
/**
 * Tool Router direct_tools session preset with OpenAI Agents
 *
 * `sessionPreset: "direct_tools"` is a shortcut for sessions where the full
 * allowed tool set is known upfront. It disables Tool Router meta/helper tools
 * by default and loads all tools allowed by the filters directly into
 * `session.tools()` and the MCP tool list.
 *
 * Usage:
 *   COMPOSIO_API_KEY=... OPENAI_API_KEY=... bun src/direct-tools-preset.ts
 */
import 'dotenv/config';
import { Composio } from '@composio/core';
import { OpenAIAgentsProvider } from '@composio/openai-agents';
import { Agent, run } from '@openai/agents';

function requireEnv(name: 'COMPOSIO_API_KEY' | 'OPENAI_API_KEY'): string {
  const value = process.env[name];
  if (!value) {
    throw new Error(`Set ${name} before running this example.`);
  }
  return value;
}

const composioApiKey = requireEnv('COMPOSIO_API_KEY');
requireEnv('OPENAI_API_KEY');

const composio = new Composio({
  apiKey: composioApiKey,
  baseURL: process.env.COMPOSIO_BASE_URL || undefined,
  provider: new OpenAIAgentsProvider(),
});

const session = await composio.create('direct-tools-example-user', {
  sessionPreset: 'direct_tools',
  toolkits: ['hackernews'],
  tools: {
    hackernews: {
      enable: ['HACKERNEWS_GET_USER'],
    },
  },
});

const tools = await session.tools();
console.log('Direct tools exposed to the agent:');
for (const tool of tools) {
  console.log(`- ${tool.name}`);
}

const agent = new Agent({
  name: 'Direct Tools Demo Agent',
  instructions:
    'Use the provided direct tools. The session is configured without Tool Router meta/helper tools.',
  model: process.env.OPENAI_MODEL ?? 'gpt-5.5',
  tools,
});

const prompt =
  process.argv.slice(2).join(' ') ||
  'Use the Hacker News tool to look up user "pg" and report their karma.';

console.log(`\nUser: ${prompt}\n`);
const result = await run(agent, prompt);
console.log(`Assistant: ${result.finalOutput}`);
